messenger.rst

```diff
@@ -633,6 +634,132 @@ config and start your workers:
 
 See the `Supervisor docs`_ for more details.
 
+If a worker dependency like your database server is down, or timeout is reached,
+you can try to add reconnect logic, or just quit the worker (catch Exception,
+PDOException or do something more fine grained):
+
+    // SomeMessageHandler.php
+    public function __invoke(SomeMessage $message): void
+    {
+        try {
+            $thing = $this->someRepository->find($message->getThingId());
+            if (null === $thing) {
+                return;
+            }
+            $this->thingService->doStuff($thing);
+        } catch (\Exception $e) {
+            echo 'Received exception "'.$e->getMessage().'" from "'.get_class($e).'". Quitting worker';
+            exit(1);
+        }
+    }
+
+But, then you can end up in a situation where the supervisor job gets into a
+FATAL (too many start retries) state. You can prevent this by wrapping the
+Symfony script with a shell script:
+
+.. code-block:: bash
+
+    #!/bin/bash
+
+    # Supervisor sends TERM to services when stopped.
+    # This wrapper has to pass the signal to it's child.
+    # Note that we send TERM (graceful) instead of KILL (immediate).
+    _term() {
+        echo "[GOT TERM, SIGNALING CHILD]"
+        kill -TERM $child 2>/dev/null
+        exit 1
+    }
+
+    trap _term SIGTERM
+
+    # Execute console.php with whatever arguments were specified to this script
+    "$@" &
+    child=$!
+    wait "$child"
+
+    # Delay to prevent supervisor from restarting too fast on failure
+    sleep 30
+
+    # Return 1 so supervisor will restart the script
```

Comment:
wouldn't it be better to exit with the Symfony exit code instead ?

Reply:
giving a 1 is reliable in the way that we know 1 will make sure supervisor restarts it. What advantages do you see when we would return the symfony exit code here?

Reply:
the wrapper will be removed